src/notifications/pollNotifications.ts

```diff
@@ -0,0 +1,39 @@
+import * as vscode from "vscode";
+import {
+  getNotifications,
+  Notification,
+  sendNotificationAction,
+} from "../binary/requests/notifications";
+import executeNotificationAction from "./executeNotificationAction";
+import { BINARY_NOTIFICATION_POLLING_INTERVAL } from "../consts";
+
+export default function pollNotifications(): void {
+  setInterval(
+    () => void doPollNotifications(),
+    BINARY_NOTIFICATION_POLLING_INTERVAL
+  );
+}
+
+async function doPollNotifications(): Promise<void> {
+  const notifications = await getNotifications();
+
+  if (!notifications || !notifications.notifications) {
+    return;
+  }
+
+  notifications.notifications.forEach(
+    (notification) => void handleNotification(notification)
+  );
+}
+
+async function handleNotification(notification: Notification): Promise<void> {
+  return vscode.window
```

Comment:
let's add a `once` wrapper on top of it with a notification.id as a key

Reply:
Why? That sounds like a hack that will bite us in the ass in the future